### PR TITLE
[MB-5926,MB-5948, MB-5951, MB-6016] Parse/test DTOD response and integrate into GHC planner

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR expands on the work started in #5508 to connect to the DTOD SOAP API.  At a high level, this PR adds the following:
- parses the response from DTOD
- adds unit tests that use a mocked DTOD API
- integrates DTOD into the GHC planner skeleton from #5501 

## Reviewer Notes

I ran into several circular dependency problems that I had to work around.

One prevented me from keeping this in the services package (since something else in services already refers back to the `route` package).  To deal with that one, I moved the DTOD work into the `route` package itself (we had a similar problem with the RM integration).  This should be OK since the planner is really a service for the entire app.  I think it even makes some of the initialization more straightforward too.

The other prevented me from generating a mock for `SoapCaller` in the `mocks` directory because a different mock referred back to the `route` package already.  To work around this, I created a `ghcmocks` directory/package instead and placed `SoapCaller` in it.

If there's another straightforward way to handle either of these, please let me know.  I didn't want to get into a huge refactoring effort in this PR since we may have to refactor unrelated code to fix this without moving DTOD-related code.

Also, I noticed there were some old mocks that were in the `pkg/services/mocks` directory as I cleaned up one of mine in this PR after the refactor.  I am deleting those as well in this PR.

## Setup

The GHC planner is not hooked up anywhere yet.  That is coming in a future story.

If you haven't already set up your `envrc.local` before with DTOD-related environment variables, then see the `Setup local env` section of #5508.

You can `make server_test` and see that the mocked SOAP API works.

To see a real call go through, you could temporarily edit `serve.go` and add in the following lines after the GHC planner is instantiated:

```
distance, err := ghcRoutePlanner.Zip5TransitDistance("05030", "05091")
logger.Debug("DTOD distance", zap.Int("distance", distance))
```

When you rebuild (`rm bin\milmove && make server_build`) and then run your server (`make server_run`), you should see a couple of debug-level statements related to distance in the log output.  Change the zips as desired to get different results.

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5926) for this change

### CODE REVIEW

File: pkg/route/ghc_planner.go [hunk 1]
```diff
@@ -52,9 +57,12 @@ func (p *ghcPlanner) Zip3TransitDistance(source string, destination string) (int
 }
 
 // NewGHCPlanner constructs and returns a Planner for GHC routing.
-func NewGHCPlanner(db *pop.Connection, logger Logger) Planner {
+func NewGHCPlanner(logger Logger, db *pop.Connection, soapClient SoapCaller, dtodUsername string, dtodPassword string) Planner {
```
Comment (pearl-truss) [inline pkg/route/ghc_planner.go:60]: The GHC Planner could be called just for Rand McNally,  in that case will it know the dtodUsername and password? Those seem to be coming from the CLI?

File: pkg/route/dtod_zip5_distance.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
+package route
+
+/*******************************************
```
Comment (pearl-truss) [inline pkg/route/dtod_zip5_distance.go:3]: Thanks, this is great documentation to have in this file

File: pkg/route/ghc_planner.go (hunk 1 above)
Comment (gidjin) [reply pkg/route/ghc_planner.go:60]: The dtodUsername and password are stored in chamber and are passed via environment variables. I believe you can override that system with CLI flags to the milmove serve command.

**Review by pearl-truss — COMMENTED** [review]
After going through the set up steps and re-running my server I get back a distance of 0 for the distance between these zips "05030" and "05091"

`2020-12-23T21:01:31.306Z        DEBUG   milmove/serve.go:585    DTOD distance   {"git_branch": "rr-MB-5926-dtod-response", "git_commit": "d23100e3de7cbf043a74fcfb2d54060059d40f86", "distance": 0}`

File: pkg/route/ghc_planner.go (hunk 1 above)
Comment (gidjin) [reply pkg/route/ghc_planner.go:60]: That said this is typical of our other such secrets for the HERE api for example, as I recall.

Comment (reggieriser) [reply pkg/route/ghc_planner.go:60]: Right, the real GHC planner is instantiated at startup of the web server and configured for both the zip3 (RM) or zip5 (DTOD) calculations.  Similarly, DTOD calls don't need the DB connection that RM does (for now), but we need to include it as part of the planner setup.

Comment (pearl-truss) [reply pkg/route/ghc_planner.go:60]: "real GHC planner is instantiated at startup" okay that makes sense!

**Comment by reggieriser** [comment]
> After going through the set up steps and re-running my server I get back a distance of 0 for the distance between these zips "05030" and "05091"

Oh, I wonder if you don't have the `env.local` variables related to DTOD yet.  I just added a note about this to the setup instructions.  See if that fixes the problem.

**Review by gidjin — APPROVED** [review]
LGTM, Tried this out and seems to return correct distances for the values I tried.

File: pkg/route/dtod_zip5_distance.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	params := gosoap.Params{
+		"DtodRequest": map[string]interface{}{
+			"AuthToken": map[string]interface{}{
+				"Username": d.username,
+				"Password": d.password,
+			},
+			"UserRequest": map[string]interface{}{
+				"Function": "Distance",
+				// TODO: Default RouteType is PcsTdyTravel, but CommercialPersonalProperty seems better. Verify.
+				"RouteType": "CommercialPersonalProperty",
```
Comment (jacquelineIO) [inline pkg/route/dtod_zip5_distance.go:96]: HHGs/PPMs moves are PCS's which stand for "permanent change of station" that's also what is selected when you onboard as the service member if I recall correctly. Better to verify, but that's why I left it as default. "CommericalPersonalProperty" to me sounds like a true commercial move and not a DoD sponsored move.

Comment (jacquelineIO) [reply pkg/route/dtod_zip5_distance.go:96]: TDY is also a DoD term for temporary duty (which is basically business travel)

**Pull request merged**